.github/workflows/publish.yml

```diff
@@ -1,34 +1,54 @@
 name: Build & publish module to registry
 
 on:
-  release:
-    types: [published]
+  push:
+    branches:
+      - '*'
+  pull_request:
+    branches:
+      - '*'
 
 jobs:
   publish:
     strategy:
       matrix:
         platform: [linux/amd64, linux/arm64, android/arm64]
+        runner: [ubuntu-latest, buildjet-8vcpu-ubuntu-2204-arm]
+        target_os: [linux, android]
+        target_arch: [amd64, arm64]
+        include:
+          - platform: linux/amd64
+            runner: ubuntu-latest
+            target_os: linux
+            target_arch: amd64
+          - platform: linux/arm64
+            runner: buildjet-8vcpu-ubuntu-2204-arm
+            target_os: linux
+            target_arch: arm64
+          - platform: android/arm64
+            runner: buildjet-8vcpu-ubuntu-2204-arm
+            target_os: android
+            target_arch: arm64
       
-    runs-on: ${{ matrix.platform == 'linux/arm64' && 'buildjet-8vcpu-ubuntu-2204-arm' || 'ubuntu-latest' }}
+    runs-on: ${{ matrix.runner }}
 
     steps:
     - uses: actions/checkout@v3
 
     - name: Set up Go
       uses: actions/setup-go@v2
       with:
-        go-version: 1.21.13
+        go-version: 1.23
```

Comment:
canon requires 1.2.3

It's not the version we use to build the module (inside of canon)